src/test/java/org/breedinginsight/services/geno/impl/GigwaGenoServiceImplIntegrationTest.java

```diff
@@ -0,0 +1,616 @@
+                                                          .count());
+        } catch (ApiException e) {
+            System.err.println(e.getMessage());
+            System.err.println(e.getResponseBody());
+            throw e;
+        }
+    }
+
+    @Test
+    public void testFetchGermplasmGenotype() throws AuthorizationException, ApiException, DoesNotExistException {
+        UUID programId = UUID.fromString("8b667063-480b-4b0a-862c-7eaa651dda28");
+        String programKey = "TESTFETCH";
+        UUID expId = UUID.randomUUID();
+        UUID importId = UUID.randomUUID();
+        assertTimeout(Duration.of(2, ChronoUnit.MINUTES), () -> uploadGenoData(programId, programKey, expId, importId), "Upload did not complete within the time period");
+
+        BrAPIGermplasm germplasm = new BrAPIGermplasm().germplasmDbId(UUID.randomUUID().toString()).germplasmName("Test Germ");
+
+        BrAPITrial trial = new BrAPITrial().externalReferences(List.of(new BrAPIExternalReference().referenceSource(Utilities.generateReferenceSource(referenceSource, ExternalReferenceSource.TRIALS))
+                                                                                                   .referenceID(UUID.randomUUID()
+                                                                                                                    .toString())));
+        doReturn(List.of(trial)).when(trialDAO).getTrials(any(UUID.class));
+
+        doAnswer(invocation -> {
+            Object searchObject = invocation.getArgument(2);
+            if(searchObject instanceof BrAPIObservationUnitSearchRequest) {
+                return List.of(new BrAPIObservationUnit().observationUnitName("USDAMSP1_A01"));
+            } else {
+                return invocation.callRealMethod();
+            }
+        }).when(brAPIDAOUtil)
+          .search(any(Function.class), any(Function3.class), any());
+
+        doAnswer(InvocationOnMock::callRealMethod).when(brAPIDAOUtil).searchWithToken(any(Function.class), any(Function3.class), any());
+
+        doReturn(new BrAPIClient("", 300000)).when(programDAO).getCoreClient(any(UUID.class));
+        doReturn(new BrAPIClient("", 300000)).when(programDAO).getPhenoClient(any(UUID.class));
+
+        GermplasmGenotype germplasmGenotype = gigwaGenoStorageService.retrieveGenotypeData(programId, germplasm);
+
+        assertNotNull(germplasmGenotype);
+        assertFalse(germplasmGenotype.getCalls().isEmpty());
+        assertFalse(germplasmGenotype.getCallSets().isEmpty());
+        assertFalse(germplasmGenotype.getVariants().isEmpty());
+    }
+
+    @Test
+    public void testSubmitValidFile() throws IOException, ApiException {
+        UUID programId = UUID.fromString("29162e85-e739-4f19-9fd0-0c377ed59956");
+        String programKey = "TESTSUBMITVALID";
+        UUID expId = UUID.randomUUID();
+
+        //TODO setup mocks for OUs
```

Comment:
Is this still needed to be implemented for this card?

Reply:
This has been implemented.  Removed TODO